### PR TITLE
Structured Logging migration: modify eviction  part logs of kubelet.

### PR DESCRIPTION
**Pull request by yangjunmyfm192085**
Signed-off-by: JunYang <yang.jun22@zte.com.cn>

#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Structured Logging migration: modify eviction  part logs of kubelet.
xref:
https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1602-structured-logging
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Migrate `pkg/kubelet/(eviction)` to structured logging
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/kubelet/eviction/eviction_manager.go [hunk 1]
```diff
@@ -196,7 +196,7 @@ func (m *managerImpl) Start(diskInfoProvider DiskInfoProvider, podFunc ActivePod
 	go func() {
 		for {
 			if evictedPods := m.synchronize(diskInfoProvider, podFunc); evictedPods != nil {
-				klog.Infof("eviction manager: pods %s evicted, waiting for pod to be cleaned up", format.Pods(evictedPods))
+				klog.InfoS("Eviction manager: pods evicted, waiting for pod to be cleaned up", "pods", format.Pods(evictedPods))
```
Comment (SergeyKanzhelev) [inline pkg/kubelet/eviction/eviction_manager.go:199]: is it expected that on line 168 the `klog.KObj(attrs.Pod)` used and here is `format.Pods` is used?

File: pkg/kubelet/eviction/eviction_manager.go [hunk 2]
```diff
@@ -176,15 +176,15 @@ func (m *managerImpl) Admit(attrs *lifecycle.PodAdmitAttributes) lifecycle.PodAd
...
+		klog.InfoS(message)
 		m.synchronize(diskInfoProvider, podFunc)
 	}
 	if m.config.KernelMemcgNotification {
 		for _, threshold := range m.config.Thresholds {
 			if threshold.Signal == evictionapi.SignalMemoryAvailable || threshold.Signal == evictionapi.SignalAllocatableMemoryAvailable {
 				notifier, err := NewMemoryThresholdNotifier(threshold, m.config.PodCgroupRoot, &CgroupNotifierFactory{}, thresholdHandler)
 				if err != nil {
-					klog.Warningf("eviction manager: failed to create memory threshold notifier: %v", err)
+					klog.InfoS("Eviction manager: failed to create memory threshold notifier", "err", err)
```
Comment (SergeyKanzhelev) [inline pkg/kubelet/eviction/eviction_manager.go:187]: why warning is replaced with Informational?

File: pkg/kubelet/eviction/eviction_manager.go [hunk 3]
```diff
@@ -252,15 +252,15 @@ func (m *managerImpl) synchronize(diskInfoProvider DiskInfoProvider, podFunc Act
...
+		klog.ErrorS(err, "Eviction manager: failed to get summary stats")
 		return nil
 	}
 
 	if m.clock.Since(m.thresholdsLastUpdated) > notifierRefreshInterval {
 		m.thresholdsLastUpdated = m.clock.Now()
 		for _, notifier := range m.thresholdNotifiers {
 			if err := notifier.UpdateThreshold(summary); err != nil {
-				klog.Warningf("eviction manager: failed to update %s: %v", notifier.Description(), err)
+				klog.InfoS("Eviction manager: failed to update notifier", "notifier", notifier.Description(), "err", err)
```
Comment (SergeyKanzhelev) [inline pkg/kubelet/eviction/eviction_manager.go:263]: warning replaced by Informational?

File: pkg/kubelet/eviction/eviction_manager.go [hunk 4]
```diff
@@ -296,7 +296,7 @@ func (m *managerImpl) synchronize(diskInfoProvider DiskInfoProvider, podFunc Act
 	// node conditions report true if it has been observed within the transition period window
 	nodeConditions = nodeConditionsObservedSince(nodeConditionsLastObservedAt, m.config.PressureTransitionPeriod, now)
 	if len(nodeConditions) > 0 {
-		klog.V(3).Infof("eviction manager: node conditions - transition period not met: %v", nodeConditions)
+		klog.V(3).InfoS("Eviction manager: node conditions - transition period not met", "nodeCondition", nodeConditions)
```
Comment (SergeyKanzhelev) [inline pkg/kubelet/eviction/eviction_manager.go:299]: how the `nodeConditions`  would be formatted? Will `%v` be applied by default?

File: pkg/kubelet/eviction/eviction_manager.go [hunk 5]
```diff
@@ -336,36 +336,36 @@ func (m *managerImpl) synchronize(diskInfoProvider DiskInfoProvider, podFunc Act
 	if !foundAny {
 		return nil
 	}
-	klog.Warningf("eviction manager: attempting to reclaim %v", resourceToReclaim)
+	klog.InfoS("Eviction manager: attempting to reclaim", "resource", resourceToReclaim)
```
Comment (SergeyKanzhelev) [inline pkg/kubelet/eviction/eviction_manager.go:339]: again, warning replaced with informational. I will not comment further on this

File: pkg/kubelet/eviction/eviction_manager.go (hunk 1 above)
Comment (yangjunmyfm192085) [reply pkg/kubelet/eviction/eviction_manager.go:199]: I want to use klog.KObj(attrs.Pod), But I think evictedPods(multiple pods) is not work here,so I keep it not change.

File: pkg/kubelet/eviction/eviction_manager.go (hunk 3 above)
Comment (yangjunmyfm192085) [reply pkg/kubelet/eviction/eviction_manager.go:263]: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md

File: pkg/kubelet/eviction/eviction_manager.go (hunk 2 above)
Comment (yangjunmyfm192085) [reply pkg/kubelet/eviction/eviction_manager.go:187]: I modified it according to the following rules
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md

File: pkg/kubelet/eviction/eviction_manager.go (hunk 4 above)
Comment (yangjunmyfm192085) [reply pkg/kubelet/eviction/eviction_manager.go:299]: I modified it according to the following rules
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md

File: pkg/kubelet/eviction/eviction_manager.go (hunk 5 above)
Comment (yangjunmyfm192085) [reply pkg/kubelet/eviction/eviction_manager.go:339]: Thak you very much.
I modified it according to the following rules,Help me see if it works?

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md

File: pkg/kubelet/eviction/eviction_manager.go (hunk 1 above)
Comment (ehashman) [reply pkg/kubelet/eviction/eviction_manager.go:199]: @serathius what should we be doing in the case of multiple pods like this? Do you think this is ok?

File: pkg/kubelet/eviction/eviction_manager.go (hunk 3 above)
Comment (ehashman) [reply pkg/kubelet/eviction/eviction_manager.go:263]: This is correct---"warning" log messages are being deprecated.

**Review by ehashman — COMMENTED** [review]
This is great, thanks, just had one question.

/cc @serathius

File: pkg/kubelet/eviction/eviction_manager.go (hunk 2 above)
Comment (SergeyKanzhelev) [reply pkg/kubelet/eviction/eviction_manager.go:187]: hm, interesting. thank you!

File: pkg/kubelet/eviction/eviction_manager.go (hunk 1 above)
Comment (serathius) [reply pkg/kubelet/eviction/eviction_manager.go:199]: I think we should migrate it to `klog.KObj` to support Json and make changes in formatting easier. We didn't add specific methods for multiple pods as we weren't sure how often this case will came up. 

This is the first case using, so for now let's just implement a simple for loop.

File: pkg/kubelet/eviction/eviction_manager.go (hunk 5 above)
Comment (serathius) [inline pkg/kubelet/eviction/eviction_manager.go:339]:
```suggestion
	klog.InfoS("Eviction manager: attempting to reclaim", "resourceName", resourceToReclaim)
```

File: pkg/kubelet/eviction/eviction_manager.go [hunk 6]
```diff
@@ -336,36 +336,36 @@ func (m *managerImpl) synchronize(diskInfoProvider DiskInfoProvider, podFunc Act
...
 
 	// record an event about the resources we are now attempting to reclaim via eviction
 	m.recorder.Eventf(m.nodeRef, v1.EventTypeWarning, "EvictionThresholdMet", "Attempting to reclaim %s", resourceToReclaim)
 
 	// check if there are node-level resources we can reclaim to reduce pressure before evicting end-user pods.
 	if m.reclaimNodeLevelResources(thresholdToReclaim.Signal, resourceToReclaim) {
-		klog.Infof("eviction manager: able to reduce %v pressure without evicting pods.", resourceToReclaim)
+		klog.InfoS("Eviction manager: able to reduce resource pressure without evicting pods.", "resource", resourceToReclaim)
```
Comment (serathius) [inline pkg/kubelet/eviction/eviction_manager.go:346]:
```suggestion
		klog.InfoS("Eviction manager: able to reduce resource pressure without evicting pods.", "resourceName", resourceToReclaim)
```

File: pkg/kubelet/eviction/eviction_manager.go [hunk 7]
```diff
@@ -336,36 +336,36 @@ func (m *managerImpl) synchronize(diskInfoProvider DiskInfoProvider, podFunc Act
...
 		return nil
 	}
 
-	klog.Infof("eviction manager: must evict pod(s) to reclaim %v", resourceToReclaim)
+	klog.InfoS("Eviction manager: must evict pod(s) to reclaim", "resource", resourceToReclaim)
```
Comment (serathius) [inline pkg/kubelet/eviction/eviction_manager.go:350]:
```suggestion
	klog.InfoS("Eviction manager: must evict pod(s) to reclaim", "resourceName", resourceToReclaim)
```

File: pkg/kubelet/eviction/eviction_manager.go [hunk 8]
```diff
@@ -336,36 +336,36 @@ func (m *managerImpl) synchronize(diskInfoProvider DiskInfoProvider, podFunc Act
...
-		klog.Errorf("eviction manager: eviction thresholds have been met, but no pods are active to evict")
+		klog.ErrorS(nil, "Eviction manager: eviction thresholds have been met, but no pods are active to evict")
 		return nil
 	}
 
 	// rank the running pods for eviction for the specified resource
 	rank(activePods, statsFunc)
 
-	klog.Infof("eviction manager: pods ranked for eviction: %s", format.Pods(activePods))
+	klog.InfoS("Eviction manager: pods ranked for eviction", "pods", format.Pods(activePods))
```
Comment (serathius) [inline pkg/kubelet/eviction/eviction_manager.go:368]: Same us above

File: pkg/kubelet/eviction/eviction_manager.go [hunk 9]
```diff
@@ -400,15 +400,15 @@ func (m *managerImpl) waitForPodsCleanup(podCleanedUpFunc PodCleanedUpFunc, pods
...
+			klog.InfoS("Eviction manager: timed out waiting for pods to be cleaned up", "pods", format.Pods(pods))
 			return
 		case <-ticker.C():
 			for i, pod := range pods {
 				if !podCleanedUpFunc(pod) {
 					break
 				}
 				if i == len(pods)-1 {
-					klog.Infof("eviction manager: pods %s successfully cleaned up", format.Pods(pods))
+					klog.InfoS("Eviction manager: pods successfully cleaned up", "pods", format.Pods(pods))
```
Comment (serathius) [inline pkg/kubelet/eviction/eviction_manager.go:411]: ditto

File: pkg/kubelet/eviction/eviction_manager.go [hunk 10]
```diff
@@ -422,14 +422,14 @@ func (m *managerImpl) reclaimNodeLevelResources(signalToReclaim evictionapi.Sign
 	for _, nodeReclaimFunc := range nodeReclaimFuncs {
 		// attempt to reclaim the pressured resource.
 		if err := nodeReclaimFunc(); err != nil {
-			klog.Warningf("eviction manager: unexpected error when attempting to reduce %v pressure: %v", resourceToReclaim, err)
+			klog.InfoS("Eviction manager: unexpected error when attempting to reduce resource pressure", "resource", resourceToReclaim, "err", err)
```
Comment (serathius) [inline pkg/kubelet/eviction/eviction_manager.go:425]:
```suggestion
			klog.InfoS("Eviction manager: unexpected error when attempting to reduce resource pressure", "resourceName", resourceToReclaim, "err", err)
```

File: pkg/kubelet/eviction/helpers.go [hunk 11]
```diff
@@ -772,9 +772,9 @@ func debugLogObservations(logPrefix string, observations signalObservations) {
 	}
 	for k, v := range observations {
 		if !v.time.IsZero() {
-			klog.Infof("eviction manager: %v: signal=%v, available: %v, capacity: %v, time: %v", logPrefix, k, v.available, v.capacity, v.time)
+			klog.InfoS("Eviction manager:", "log", logPrefix, "signal", k, "resource", v.available, "capacity", v.capacity, "time", v.time)
```
Comment (serathius) [inline pkg/kubelet/eviction/helpers.go:775]:
```suggestion
			klog.InfoS("Eviction manager:", "log", logPrefix, "signal", k, "resourceName", v.available, "capacity", v.capacity, "time", v.time)
```

File: pkg/kubelet/eviction/helpers.go [hunk 12]
```diff
@@ -772,9 +772,9 @@ func debugLogObservations(logPrefix string, observations signalObservations) {
...
 		} else {
-			klog.Infof("eviction manager: %v: signal=%v, available: %v, capacity: %v", logPrefix, k, v.available, v.capacity)
+			klog.InfoS("Eviction manager:", "log", logPrefix, "signal", k, "resource", v.available, "capacity", v.capacity)
```
Comment (serathius) [inline pkg/kubelet/eviction/helpers.go:777]:
```suggestion
			klog.InfoS("Eviction manager:", "log", logPrefix, "signal", k, "resourceName", v.available, "capacity", v.capacity)
```

**Pull request merged**